enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Filter/FilterDropdown/FilterDropdown.tsx

```diff
@@ -0,0 +1,113 @@
+import { MultiStepPopover } from "embedding-sdk/components/private/util/MultiStepPopover";
+import type { UpdateQueryHookProps } from "metabase/query_builder/hooks";
+import { getFilterItems } from "metabase/querying/filters/components/FilterPanel/utils";
+import type { FilterColumnPickerProps } from "metabase/querying/filters/components/FilterPicker/FilterColumnPicker";
+import type * as Lib from "metabase-lib";
+
+import { useInteractiveQuestionContext } from "../../../context";
+import { ToolbarButton } from "../../util/ToolbarButton";
+import { FilterPicker } from "../FilterPicker/FilterPicker";
+
+type FilterProps = Pick<
+  FilterColumnPickerProps,
+  "withColumnItemIcon" | "withColumnGroupIcon" | "withCustomExpression"
+>;
+
+const FilterDropdownInner = ({
+  query,
+  withColumnItemIcon,
+}: UpdateQueryHookProps & FilterProps) => {
+  const filters = useMemo(() => getFilterItems(query), [query]);
+
+  const [step, setStep] = useState<"list" | "picker">("list");
+
+  const [opened, { close, toggle }] = useDisclosure(false, {
+    onOpen: () => {
+      setStep(filters.length === 0 ? "picker" : "list");
+    },
+  });
+
+  const [selectedFilter, setSelectedFilter] = useState<
+    SDKFilterItem | undefined
+  >();
+
+  const onSelectFilter = (item?: SDKFilterItem) => {
+    setSelectedFilter(item);
+    setStep("picker");
+  };
+
+  const label = match(filters.length)
+    .with(0, () => t`Filter`)
+    .with(1, () => t`1 filter`)
+    .otherwise(value => jt`${value} filters`);
+
+  const onRemoveItem = (item: SDKFilterItem) => {
+    item.onRemoveFilter();
+    if (filters.length === 1) {
+      close();
+    }
+  };
+
+  return (
+    <MultiStepPopover currentStep={step} opened={opened} onClose={close}>
+      <MultiStepPopover.Target>
+        <ToolbarButton
+          label={label}
+          icon="filter"
+          isHighlighted={filters.length > 0}
+          onClick={toggle}
+        />
+      </MultiStepPopover.Target>
+
+      <MultiStepPopover.Step value="list">
+        <FilterBadgeList
+          onAddItem={onSelectFilter}
+          onSelectItem={onSelectFilter}
+          onRemoveItem={onRemoveItem}
+        />
+      </MultiStepPopover.Step>
+
+      <MultiStepPopover.Step value="picker">
+        <FilterPicker
+          filterItem={selectedFilter}
+          withIcon={withColumnItemIcon}
+          onClose={() => setStep("list")}
+        />
+      </MultiStepPopover.Step>
+    </MultiStepPopover>
+  );
+};
+
+export const FilterDropdown = ({
+  withColumnItemIcon,
+  withColumnGroupIcon,
+  withCustomExpression,
+}: FilterProps) => {
+  const { question, updateQuestion } = useInteractiveQuestionContext();
+
+  if (!question) {
+    return null;
+  }
+
+  const onQueryChange = (query: Lib.Query) => {
+    updateQuestion(question.setQuery(query), { run: true });
+  };
+
+  return (
+    <FilterDropdownInner
```

Comment:
This component only accepts 2 props, so the rest wouldn't do anything, so we should remove them.

https://github.com/metabase/metabase/blob/61de2e5c1651a058ab7f783e010878dfcc77d242/enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Filter/FilterDropdown/FilterDropdown.tsx#L23-L26